src/backend/aspen/app/views/sample.py

```diff
@@ -150,57 +149,39 @@ def _format_lineage(sample: Sample) -> dict[str, Any]:
 @application.route("/api/sequences", methods=["GET"])
 @requires_auth
 def prepare_sequences_download():
-    with session_scope(application.DATABASE_INTERFACE) as db_session:
-        profile = session["profile"]
-        user = (
-            get_usergroup_query(db_session, profile["user_id"])
-            .options(joinedload(User.group).joinedload(Group.can_see))
-            .one()
-        )
-        request_data = request.get_json()
-        sample_ids = request_data["requested_sequences"]["sample_ids"]
-
-        # query for samples
-        all_samples: Iterable[Sample] = (
-            db_session.query(Sample)
-            .filter(
-                and_(
-                    Sample.uploaded_pathogen_genome != None,  # noqa: E711
-                    or_(
-                        Sample.private_identifier.in_(sample_ids),
-                        Sample.public_identifier.in_(sample_ids),
+    # stream output file
+    user = g.auth_user
+    fasta_filename = f"{user.group.name}_sample_sequences.fasta"
+    request_data = request.get_json()
+    sample_ids = request_data["requested_sequences"]["sample_ids"]
+
+    @stream_with_context
+    def stream_samples(cansee_groups_private_identifiers):
+        with session_scope(application.DATABASE_INTERFACE) as db_session:
```

Comment:
What's the reason for having to create a new db_session here instead of passing in the existing one?

Reply:
As far as I know, ~all API endpoints will require DB access so creating the seeion here declutters the endpoint code a little bit. I think the main drawback to managing this centrally is that it can make advanced db session management (explicitly closing/reopening sessions) more confusing if any endpoints need to do that.

Reply:
prepare_sequences_download() returns this generator, and then the requires_auth() generator returns it, closing the db session scope we opened in requires_auth. Even our request context would normally be out of scope, except that we're using flask's [stream_with_context](https://flask.palletsprojects.com/en/2.0.x/patterns/streaming/#streaming-with-context) decorator to keep it around.